cmake/Dependencies.cmake

```
@@ -1248,6 +1248,21 @@ if(ANDROID)
   list(APPEND Caffe2_DEPENDENCY_LIBS log)
 endif()
 
+# ---[ Kernel asserts
+# Kernel asserts are enabled by default for CUDA and disabled for ROCm.
+# For ROCm, it can be enabled by setting FORCE_ENABLE_GPU_ASSERTS
+if(USE_ROCM)
+  message(STATUS "Disabling kernel asserts for ROCm")
+  caffe2_update_option(TORCH_DISABLE_GPU_ASSERTS ON)
+else()
+  message(STATUS "Enabling kernel asserts")
+endif()
+
+if(FORCE_ENABLE_GPU_ASSERTS)
+  caffe2_update_option(TORCH_DISABLE_GPU_ASSERTS OFF)
+  message(STATUS "Forcefully enabling kernel asserts")
+endif()
```

Wouldn't something like that be cleaner?
```suggestion
if(USE_ROCM AND NOT ROCM_FORCE_ENABLE_GPU_ASSERTS)
  message(STATUS "Disabling kernel asserts for ROCm")
  caffe2_update_option(TORCH_DISABLE_GPU_ASSERTS ON)
elseif(USE_ROCM AND ROCM_FORCE_ENABLE_GPU_ASSERTS)
  message(STATUS "Forcefully enabling kernel asserts on ROCM")
endif()
```

---

Tried to keep the variable  `FORCE_ENABLE_GPU_ASSERTS` generic, will keep it ROCm specific and update the code accordingly.

---

Updated the code.